### ISSUE TITLE
Learn & Apply algorithm

### ISSUE THREAD

**Issue by littlepitta** [AOtools/soapy#130]
I have read Vidal's and your thesis paper and used Soapy to set up a learn & apply algorithm simulation. 

Here's what I have done.
I used reconstructor as LearnAndApply, 
and I put 2 DMs: TT with closed and FastPiezo with open, so that TT can work as a so-called "truth sensor".
However, I came up with an error that saying the matrix shape didn't fit. 
I think it is something to do with WFS-nxSubaps and DM-nxActuators problem, but I couldn't figure out how to fix it.

Anyway, 2 DMs didn't work, so I only used one DM which is FastPiezo.
Now this time, the simulation works fine but the value of the Strehl ratio is too low ~0.2 or so.
When I used reconstructor as MVM, the Strehl ratio was high enough like 0.5~0.6. 

To sum up, 
1. How do I use 2 DMs?
2. How do I get a fair Strehl ratio with the Learn&Apply algorithm?

Sorry for the long post. 
Thanks

**Comment by ojdf** [comment]
Hi @littlepitta, as a starting point can you post the config file that you are using? Thanks

**Comment by ojdf** [comment]
Hi, I've had a look at the LearnAndApply reconstructor. It has been a long time since it has been used so there were various updates that needed to happen. Please try out the `sh_8x8_learn&apply.yaml` config in my fork https://github.com/ojdf/soapy/tree/learn-apply-fixes 

Re: gui, I've just tested on Rocky Linux 8 and it works, are you using the most recent version of soapy?

**Comment by littlepitta** [comment]
@ojdf  Hi, thanks for the reply. 
I found your fork `sh_8x8_learn&apply.yaml`, but I have a few questions including recent changes.

1. The system changed into closed loop
2. TT DM was removed.
I thought that Learn&Apply algorithm is open loop system with "truth sensor" closed. Is that applied in yaml file? 

Thanks! 

p.s. GUI matter is not that important but...
([https://github.com/ojdf/soapy/tree/learn-apply-fixes] is this the most recent ver.?)
